demo/site/server.ts

```diff
@@ -30,7 +34,32 @@ app.prepare()
                         return;
                     }
                 }
+                if (parsedUrl.pathname?.startsWith("/assets/")) {
+                    // don't let nextjs handle public assets so we can have custom cache-control
+                    // TODO move public/* files into public/assets folder
+                    const done = finalhandler(req, res);
+                    handlePublic(req, res, done);
+                    return;
+                } else if (parsedUrl.pathname == "/favicon.ico" || parsedUrl.pathname == "/robots.txt" || parsedUrl.pathname == "/sitemap.xml") {
+                    res.setHeader("Cache-Control", "public, max-age=900");
+                    const origSetHeader = res.setHeader;
+                    res.setHeader = function (name: string, value: string | number | readonly string[]) {
+                        if (name === "cache-control" || name === "Cache-Control") {
+                            // ignore
+                            return;
+                        }
+                        return origSetHeader.call(this, name, value);
+                    };
+                } else if (
+                    req.method === "GET" &&
+                    !parsedUrl.pathname?.startsWith("/_next/") &&
+                    !parsedUrl.pathname?.includes("api/") // TODO: find a better way to exclude api routes
```

Comment:
Do you have any ideas to achieve this?

Also, Next caches Route Handlers with GET by default, could this lead to problems?

https://nextjs.org/docs/app/building-your-application/routing/route-handlers#caching

Reply:
> Do you have any ideas to achieve this?

Unfortunately not. 

> Also, Next caches Route Handlers with GET by default,

whatever "caches" means. Data-Cache? Full-Route-Cache? In my tests they didn't output Cache-Control headers.

Reply:
I guess some of their "caches"

> Unfortunately not.

Very primitive idea: Scan the app/ directory for all route.ts files and build an array of route handler paths

Reply:
we decided to use the default next behavior, so detecting api routes is not needed